cirq/linalg/decompositions.py

```diff
@@ -514,6 +528,120 @@ def interaction_matrix(m: np.ndarray, c: float) -> np.ndarray:
+        the following two constraints:
+
+            0 <= abs(z) <= y <= x <= 1
+            if x = 1 then z >= 0
+
+        The wireframe includes lines along the surface of the space at z=0.
+
+    Args:
+        interactions: An iterable of two qubit unitary interactions. Each
+            interaction can be specified as a raw 4x4 unitary matrix, or an
+            object with a 4x4 unitary matrix according to `cirq.unitary` (
+            (e.g. `cirq.CZ` or a `cirq.KakDecomposition` or a `cirq.Circuit`
+            over two qubits).
+        include_frame: Determines whether or not to draw the kak space
+            wireframe. If not set. Defaults to `True`.
+        ax: A matplotlib 3d axes object to plot into. If not specified, a new
+            figure is created.
+        show: Whether or not to call `matplotlib.pyplot.show()`. Defaults to
+            `False`.
+        kwargs: Arguments forwarded into the call to `scatter` that plots the
+            points. Working arguments include color `c='blue'`, scale `s=2`,
+            labelling `label="theta=pi/4"`, etc. For reference see the
+            `matplotlib.pyplot.scatter` documentation:
+            https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.scatter.html
+
+    Returns:
+        The matplotlib 3d axes object that was plotted into.
+
+    Examples:
+        >>> ax = None
+        >>> for y in np.linspace(0, 0.5, 4):
+        ...     a, b = cirq.LineQubit.range(2)
+        ...     circuits = [
+        ...         cirq.Circuit.from_ops(
+        ...             cirq.CZ(a, b)**0.5,
+        ...             cirq.X(a)**y, cirq.X(b)**x,
+        ...             cirq.CZ(a, b)**0.5,
+        ...             cirq.X(a)**x, cirq.X(b)**y,
+        ...             cirq.CZ(a, b) ** 0.5,
+        ...         )
+        ...         for x in np.linspace(0, 1, 25)
+        ...     ]
+        ...     ax = cirq.scatter_plot_normalized_kak_interaction_coefficients(
+        ...         circuits,
+        ...         s=1,
+        ...         ax=ax,
+        ...         include_frame=ax is None,
+        ...         label=f'y={y:0.2f}')
+        >>> _ = ax.legend()
+        >>> import matplotlib.pyplot as plt
+        >>> plt.show()
+    """
+    if ax is None:
+        fig = plt.figure()
+        ax = fig.add_subplot(projection='3d')
+
+    def coord_transform(
+            pts: Iterable[Tuple[float, float, float]]
+    ) -> Tuple[Iterable[float], Iterable[float], Iterable[float]]:
+        xs, ys, zs = zip(*pts)
+        return xs, zs, ys
+
+    if include_frame:
+        envelope = [
+            (0, 0, 0),
+            (1, 1, 1),
+            (1, 1, -1),
+            (0, 0, 0),
+            (1, 1, 1),
+            (1, 0, 0),
+            (0, 0, 0),
+            (1, 1, -1),
+            (1, 0, 0),
+            (0, 0, 0),
+            (1, 0, 0),
+            (1, 1, 0),
+            (0, 0, 0),
+        ]
+        ax.plot(*coord_transform(envelope), c='black', linewidth=1)
```

Comment:
Could you add labels to the points that correspond to well-known gates like CNOT/CZ, iSWAP, SWAP etc? I think this would be helpful to new users and people who use cirq to learn about KAK.

Reply:
I think putting text at these points would clutter the plot. They are already highlighted in the sense that they are at corners or intersections of blue lines. The SWAP is the top left = top right corner. CZ is the bottom of the vertical triangle. ISWAP is the middle of the top of the vertical triangle.

I'll see if I can draw something i nthe doc string.

Reply:
How's this

```
        The space is a prism with the identity at the origin, a crease along
        y=z=0 leading to the CZ/CNOT at x=1 and a vertical triangular face that
        contains the iswap at x=y=1,z=0 and the swap at x=y=z=1:

                                 (x=1,y=1,z=0)   
                             swap___iswap___swap (x=1,y=1,z=-1)
                               _/\    |    /
                             _/   \   |   /
                           _/      \  |  /
                         _/         \ | /
                       _/            \|/
        (x=0,y=0,z=0) I---------------CZ (x=1,y=0,z=0)
```

Reply:
How about using colored points at the vertices and adding a legend for the color mapping? You'd have to somehow avoid the color collisions with the input data, though.

Reply:
The ASCII art also works (didn't see it before posting previous comment).